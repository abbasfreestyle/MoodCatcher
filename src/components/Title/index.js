import React from 'react';
import PropTypes from 'prop-types';

import { View, Text, StyleSheet } from 'react-native';

import theme from 'theme';

const styles = StyleSheet.create({
  container: {
    margin: 20
  },
  text: {
    color: theme.primary,
    fontWeight: '400',
    fontSize: 30
  }
});

const Title = props => {
  const { children } = props;
  return (
    <View style={styles.container}>
      <Text style={styles.text}>{children}</Text>
    </View>
  );
};

Title.propTypes = {
  children: PropTypes.string.isRequired
};

export default Title;
