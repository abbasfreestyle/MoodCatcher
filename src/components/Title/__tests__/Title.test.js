import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Title from '..';

const requiredProps = {
  children: 'Test Title'
};

const wrapper = shallow(<Title {...requiredProps} />);

describe('Title.', () => {
  describe('Snapshots.', () => {
    test('should match', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
