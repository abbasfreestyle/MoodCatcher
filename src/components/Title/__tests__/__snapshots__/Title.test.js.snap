// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title. Snapshots. should match 1`] = `
<View
  style={
    Object {
      "margin": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#44BBA4",
        "fontSize": 30,
        "fontWeight": "400",
      }
    }
  >
    Test Title
  </Text>
</View>
`;
